doc/gen_docs.go

```diff
@@ -0,0 +1,185 @@
+package doc
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+type CmdOutline struct {
+	Name          string   // full path to the command
+	Short         string   // short description of the command
+	Long          string   // long description of the command
+	UseLine       string   // full usage for a given command (including parents)
+	Example       string   // examples of how to use the command
+	Flags         string   // default values of all non-inherited flags as a string
+	FlagSlice     []string // Flags represented as a slice
+	ParentFlags   string   // default values of all inherited flags as a string
+	ParentLink    string   // rendered internal link to the parent command
+	ChildrenLinks []string // rendered internal links to the child commands as a slice
+	RelatedLinks  []string // rendered internal links to the related commands as a slice
+	CommandLink   string   // rendered internal link to the command
+	HeaderScale   int      // integer scale indicating depth of the current command
+	AutoGenTag    string   // automatically generated tag by Cobra
+}
+
+func generateCmdOutline(cmd *cobra.Command, linkHandler func(string) string) *CmdOutline {
+	name := cmd.CommandPath()
+	short := cmd.Short
+	long := cmd.Long
+	if len(long) == 0 {
+		long = short
+	}
+
+	var useLine string
+	useLine = cmd.UseLine()
+
+	var example string
+	if len(cmd.Example) > 0 {
+		example = cmd.Example
+	}
+
+	buf := new(bytes.Buffer)
+
+	var flagString string
+	flags := cmd.NonInheritedFlags()
+	flags.SetOutput(buf)
+	if flags.HasAvailableFlags() {
+		flags.PrintDefaults()
+		flagString = buf.String()
+		buf.Reset()
+	}
+
+	var flagSlice []string
+	flagSplitter := regexp.MustCompile("\n *-") // clips first '-' in flags starting the second flag
+	numberOfFlags := strings.Count(flagString, "--")
+	flagSlice = flagSplitter.Split(flagString, numberOfFlags)
+	if flagSlice[len(flagSlice)-1] == "" {
+		flagSlice = flagSlice[:len(flagSlice)-1]
+	}
+	for i, flag := range flagSlice {
+		if i > 0 {
+			flagSlice[i] = "-" + flag // add clipped '-' back to flags
+		}
+	}
+
+	var parentFlagString string
+	parentFlags := cmd.InheritedFlags()
+	parentFlags.SetOutput(buf)
+	if parentFlags.HasAvailableFlags() {
+		parentFlags.PrintDefaults()
+		parentFlagString = buf.String()
+		buf.Reset()
+	}
+
+	headerScale := 0
+	var parentLink string
+	if cmd.HasParent() {
+		parent := cmd.Parent()
+		pname := parent.CommandPath()
+		link := pname + ".md"
+		link = strings.Replace(link, " ", "_", -1)
+		buf.WriteString(fmt.Sprintf("* [%s](%s)\t - %s\n", pname, linkHandler(link), parent.Short))
```

Comment:
Why are you using a buffer in several places in this file? Seems like you're always writing to it once and then resetting it... Why not use the string that `fmt.Sprintf` returns to you?

Reply:
Ah yes another product of my simply copying the original code. I'll fix those